vendor/assets/javascripts/neat/model_editor.coffee

```diff
@@ -94,26 +94,54 @@ class window.Neat.ModelEditor extends Backbone.View
   okToSave: (attributes)->
     true
 
-  attributesFromForm: ($el)->
+  attributesFromForm: ($el, options)->
+    options = options ? {}
     attrs = {}
     $el.find('input, select, textarea').each ->
-      elem = $(this)
+      elem = $(@)
       name = elem.attr('name')
-      if name
-        elemType = elem.attr('type')
-        value = elem.val()
-
-        if name.substr(-2) == '[]'
-          name = name.substring(0, name.length - 2)
-          attrs[name] = attrs[name] || []
-          attrs[name].push elem.val()
-
-        else if elemType == 'checkbox' || elemType == 'radio'
-          attrs[name] = '' if typeof(attrs[name]) == 'undefined'
-          attrs[name] = value if elem.prop('checked')
-
-        else
-          attrs[name] = value
+      elemType = elem.attr('type')
+      value = elem.val()
+      return true unless name
+
+      # Parse out nested objects as represented in names
+      # person[address][zip]=63303 should be serialized to:
+      # person:
+      #   address:
+      #     zip: 63303
+      parts = _.without(name.split(/\[([^\]]+)\]/), '')
+      name = parts.pop()
+      isArray = false
+      while name is '[]'
+        isArray = true
+        name = parts.pop()
+      context = attrs
+      for part in parts
+        context = context[part] or (context[part] = {})
+
+      if value is null and options.stringifyNullValues
+        value = ''
+
+      # Normally, checkboxes serialize to a value if they are
+      # checked and do not if they are not checked. Alernately,
+      # we can serialize the boolean indicating the checkbox's
+      # checked-state as its value rather than its value.
+      if elemType == 'checkbox' && options.checkboxesAreBoolean
+        value = elem.prop('checked')
+        elemType = null
+
+      if (elemType == 'checkbox' || elemType == 'radio') && !elem.prop('checked')
+        return true
+
+      if isArray
+        # select with multiple=true will return
+        # an array of selected values, so we don't
+        # need to nest that array in another array
+        value = [value] unless _.isArray(value)
+        value = (context[name] || []).concat(value)
+
+      context[name] = value
+      true # Don't break out of the loop
```

Comment:
It looks like you copied-and-pasted this from Members.

I don't think it makes sense to add `options` to this method, since the calling code won't ever pass them. Those options should instead be all the way up on line 3 — options for the ModelEditor.

_But_ I'm not sure we need either of those options here — and there both off by default — so, since as of now, they'd never be used, I'd suggest removing them.